### PR TITLE
tune down the max multi-get limit to 50

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

Limit the max multi-get return number. This helps reduce the load on system for each request and risk of resources starvation. 

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-json-rpc/src/api/mod.rs [hunk 1]
```diff
@@ -39,15 +39,11 @@ mod read;
...
-/// This is equivalent to EVENT_QUERY_MAX_LIMIT in `sui-storage` crate.
-/// To avoid unnecessary dependency on that crate, we have a reference here
-/// for document purposes.
-pub const QUERY_MAX_RESULT_LIMIT: usize = 1000;
+pub const QUERY_MAX_RESULT_LIMIT: usize = 50;
 // TODOD(chris): make this configurable
 pub const QUERY_MAX_RESULT_LIMIT_CHECKPOINTS: usize = 100;
 
-pub const QUERY_MAX_RESULT_LIMIT_OBJECTS: usize = 256;
+pub const QUERY_MAX_RESULT_LIMIT_OBJECTS: usize = 50;
```
Comment (666lcz) [inline crates/sui-json-rpc/src/api/mod.rs:46]: @gegaowp , does this break the assumption on indexer by any chance?

Comment (gegaowp) [reply crates/sui-json-rpc/src/api/mod.rs:46]: @666lcz is this applied to objects and tx ? if I requested 200, will it be capped to 50 or return error?

**Comment by gegaowp** [comment]
Nvm, please update this to 50 as well
https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/src/handlers/checkpoint_handler.rs#L51

File: crates/sui-json-rpc/src/api/mod.rs (hunk 1 above)
Comment (gegaowp) [reply crates/sui-json-rpc/src/api/mod.rs:46]: nvm, added another comment for indexer consistency

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ use crate::IndexerConfig;
 const MAX_PARALLEL_DOWNLOADS: usize = 24;
 const DOWNLOAD_RETRY_INTERVAL_IN_SECS: u64 = 10;
 const DB_COMMIT_RETRY_INTERVAL_IN_MILLIS: u64 = 100;
-const MULTI_GET_CHUNK_SIZE: usize = 200;
+const MULTI_GET_CHUNK_SIZE: usize = 50;
```
Comment (666lcz) [inline crates/sui-indexer/src/handlers/checkpoint_handler.rs:51]: nit: we can probably remove this and directly use `QUERY_MAX_RESULT_LIMIT`

**Review by 666lcz — APPROVED** [review]

**Review by gegaowp — APPROVED** [review]

**Pull request merged**